core/consensus/babe/impl/babe_impl.cpp

```diff
@@ -113,13 +113,21 @@ namespace kagome::consensus::babe {
 
     if (auto epoch_res = block_tree_->getEpochDescriptor(
             last_epoch_descriptor.epoch_number, best_block_.hash);
-        epoch_res.has_value() and epoch_res.value().authorities.size() == 1
-        and epoch_res.value().authorities[0].id.id == keypair_->public_key) {
-      SL_INFO(log_, "Starting single validating node.");
-      onSynchronized();
+        epoch_res.has_value()) {
+      const auto &authorities = epoch_res.value().authorities;
+      if (authorities.size() == 1
+          and authorities[0].id.id == keypair_->public_key) {
+        SL_INFO(log_, "Starting single validating node.");
+        onSynchronized();
+        return true;
+      }
     } else {
-      current_state_ = State::WAIT_REMOTE_STATUS;
+      SL_CRITICAL(log_,
+                  "Epoch couldn't be obtained from epoch #{}, block hash {}",
+                  last_epoch_descriptor.epoch_number,
+                  best_block_.hash.toHex());
```

Comment:
Please, log error message (with `epoch_res.error().message()`) and **return false** in the end.